### PR TITLE
fix: OpenId migration failing for MySql and PostgreSql

### LINKED ISSUE #4655: BUG OpendId FK too long on MySql database
OrchardCore.Data.Migration.DataMigrationManager[0]
      Error while running migration version 0 for 'OrchardCore.OpenId.Management'.
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> MySql.Data.MySqlClient.MySqlException (0x80004005): Identifier name 'FK_OpenIdApplicationByPostLogoutRedirectUriIndex_Document_DocumentId' is too long
 ---> MySql.Data.MySqlClient.MySqlException (0x80004005): Identifier name 'FK_OpenIdApplicationByPostLogoutRedirectUriIndex_Document_DocumentId' is too long
   at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in C:\projects\mysqlconnector\src\MySqlConnector\Core\ResultSet.cs:line 49
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet() in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 127
   at MySql.Data.MySqlClient.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 340
   at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\Core\CommandExecutor.cs:line 64
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlCommand.cs:line 227
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery() in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlCommand.cs:line 75
   at Dapper.SqlMapper.ExecuteCommand(IDbConnection cnn, CommandDefinition& command, Action`2 paramReader) in C:\projects\dapper\Dapper\SqlMapper.cs:line 2797
   at Dapper.SqlMapper.ExecuteImpl(IDbConnection cnn, CommandDefinition& command) in C:\projects\dapper\Dapper\SqlMapper.cs:line 568
   at YesSql.Sql.SchemaBuilder.Execute(IEnumerable`1 statements)
   at YesSql.Sql.SchemaBuilder.CreateForeignKey(String name, String srcTable, String[] srcColumns, String destTable, String[] destColumns)
   at YesSql.Sql.SchemaBuilder.CreateReduceIndexTable(String name, Action`1 table)
   at OrchardCore.OpenId.YesSql.Migrations.OpenIdMigrations.Create() in C:\projects\orchardcore\src\OrchardCore\OrchardCore.OpenId.Core\YesSql\Migrations\OpenIdMigrations.cs:line 19
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at OrchardCore.Data.Migration.DataMigrationManager.UpdateAsync(String featureId) in C:\projects\orchardcore\src\OrchardCore\OrchardCore.Data\Migration\DataMigrationManager.cs:line 193

### PR DESCRIPTION
**Pull request by jptissot**
fix #4655 

Reduce length of reduce index table to fix of migration failing.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs [hunk 1]
```diff
@@ -10,9 +10,9 @@ public class OpenIdApplicationIndex : MapIndex
         public string ClientId { get; set; }
     }
 
-    public class OpenIdApplicationByPostLogoutRedirectUriIndex : ReduceIndex
+    public class OpenIdAppPostLogoutRedirectIndex : ReduceIndex
```
Comment (PinpointTownes) [inline src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs]: Should we remove `Post` from the class name to match the new property name?

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Migrations/OpenIdMigrations.cs [hunk 2]
```diff
@@ -12,13 +12,13 @@ public int Create()
...
                 .Column<string>(nameof(OpenIdApplicationIndex.ClientId), column => column.Unique()));
 
-            SchemaBuilder.CreateReduceIndexTable(nameof(OpenIdApplicationByPostLogoutRedirectUriIndex), table => table
-                .Column<string>(nameof(OpenIdApplicationByPostLogoutRedirectUriIndex.PostLogoutRedirectUri))
-                .Column<int>(nameof(OpenIdApplicationByPostLogoutRedirectUriIndex.Count)));
+            SchemaBuilder.CreateReduceIndexTable(nameof(OpenIdAppPostLogoutRedirectIndex), table => table
+                .Column<string>(nameof(OpenIdAppPostLogoutRedirectIndex.LogoutRedirectUri))
+                .Column<int>(nameof(OpenIdAppPostLogoutRedirectIndex.Count)));
 
             SchemaBuilder.CreateReduceIndexTable(nameof(OpenIdApplicationByRedirectUriIndex), table => table
```
Comment (PinpointTownes) [inline src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Migrations/OpenIdMigrations.cs]: Should we rename the other indexes for consistency?

**Comment by jptissot** [comment]
Without tenant prefix this works.

image

image

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs (hunk 1 above)
Comment (jptissot) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs]: Good catch

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Migrations/OpenIdMigrations.cs (hunk 2 above)
Comment (jptissot) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Migrations/OpenIdMigrations.cs]: Ok will do.

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs (hunk 1 above)
Comment (PinpointTownes) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs]: What about `OpenIdApplicationLogoutUriIndex`? Would that be too long?

Comment (jptissot) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Indexes/OpenIdApplicationIndex.cs]: I changed everything to `...App...`

**Pull request merged**